README.md

```diff
@@ -1 +1,172 @@
-# figma_layout_grid
+# Figma layout grid
+`figma_layout_grid` is a Flutter package which imitates Figma's layout grid and all its functionalities. It helps you determine whether your UI components match the design file and align properly. 
+
+## Basic usage
+
+Simply wrap a widget you wish to have layout grid rendered on top with `LayoutGrid` by passing it as a `builder` parameter. 
+
+
+```dart
+LayoutGrid(
+    ...
+    builder: (context) => Scaffold(),
+)
+```
+
+## State management
+
+By default all of the layouts are disabled. In order to manage layout grid state you may use `LayoutGridController` methods which are accessible via `builder` context. 
+
+### Rows 
+`void LayoutGridController.of(context).showRows()` - shows row layout
+
+`void LayoutGridController.of(context).hideRows()` - hides row layout
+
+`void LayoutGridController.of(context).toggleRows()` - toggles row layout visibility
+
+`bool LayoutGridController.of(context).visibleRows` - returns current row layout visibility state 
+
+### Columns 
+`void LayoutGridController.of(context).showColumns()` - shows column layout
+
+`void LayoutGridController.of(context).hideColumns()` - hides column layout
+
+`void LayoutGridController.of(context).toggleColumns()` - toggles column layout visibility
+
+`bool LayoutGridController.of(context).visibleColumns` - returns current column layout visibility state 
+
+### Grid 
+`void LayoutGridController.of(context).showGrid()` - shows grid layout
+
+`void LayoutGridController.of(context).hideGrid()` - hides grid layout
+
+`void LayoutGridController.of(context).toggleGrid()` - toggles grid layout visibility
+
+`bool LayoutGridController.of(context).visibleGrid` - returns current grid layout visibility state 
+
+## Available layouts
+`figma_layout_grid` supports three types of layout overlays - rows, columns and pixel grid.
+
+### Rows
+
+Horizontal stripes with specified height, gutter, margin, offset, arrangement and color.
+There is a possibility for fixed rows count with passing desired number in a `count` parameter, otherwise rows are rendered dynamically due to the screen limits. 
+
+Example usage
+
+```dart
+LayoutGrid(
+  ...
+  rowsParams: const RowsParams(
+    offset: 32.0,
+    height: 16.0,
+    gutter: 16.0,
+  ),
+  builder: (context) => Scaffold(),
+),
+```
+<img src="https://raw.githubusercontent.com/Iteo/figma_layout_grid/implementation/resources/rows.png" alt="rows example image" style="width:300px;"/>
```

Comment:
```suggestion
<img src="https://raw.githubusercontent.com/Iteo/figma_layout_grid/main/resources/rows.png" alt="rows example image" style="width:300px;"/>
```
otherwise it will stop working when we will remove this branch